pymongo/topology_description.py

```diff
@@ -536,8 +556,12 @@ def _update_rs_from_primary(
         return _check_has_primary(sds), replica_set_name, max_set_version, max_election_id
 
     if server_description.max_wire_version is None or server_description.max_wire_version < 17:
-        new_election_tuple = server_description.set_version, server_description.election_id
-        max_election_tuple = max_set_version, max_election_id
+        new_election_tuple: Union[
+            Tuple[Optional[int], Optional[ObjectId]], Tuple[Optional[ObjectId], Optional[int]]
+        ] = (server_description.set_version, server_description.election_id)
+        max_election_tuple: Union[
+            Tuple[Optional[int], Optional[ObjectId]], Tuple[Optional[ObjectId], Optional[int]]
+        ] = (max_set_version, max_election_id)
```

Comment:
Why do these need explicit hints? It makes this code significantly less readable.

Reply:
Um the other alternative that I see is to just type it as `Tuple` and not specify the contents. By default, mypy will assume `new_election_tuple` to be a tuple of `Optional[int]`, `Optional[OblectId]`. however, it gets reassigned to be a tuple of `Optional[ObjectId]`, `Optional[int]` which mypy then complains about. 

I've changed it to now just be `Tuple`

Reply:
Oh I see the problem is that mypy considers the two vars to be overlapping even though they are never used in the same scope. Yeah I think bare Tuple is fine.